frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/SlideOut.java

```diff
@@ -71,70 +71,72 @@ public void run(@UI Runnable doInUIThreadWhenReady) {
     val slideOutBranchIsCurrent = branchToSlideOutName.equals(gitRepository.getCurrentBranchName());
     if (slideOutBranchIsCurrent) {
       LOG.debug("Skipping (optional) local branch deletion because it is equal to current branch");
-      slideOutBranch(branchToSlideOutName);
-      VcsNotifier.getInstance(project).notifySuccess(/* displayId */ null,
-          /* title */ "",
-          getString("action.GitMachete.SlideOut.notification.title.slide-out-success.of-current.HTML").fmt(
-              branchToSlideOutName));
-      ModalityUiUtil.invokeLaterIfNeeded(ModalityState.NON_MODAL, doInUIThreadWhenReady);
+      slideOutBranch(() -> {
+        VcsNotifier.getInstance(project).notifySuccess(/* displayId */ null,
+            /* title */ "",
+            getString("action.GitMachete.SlideOut.notification.title.slide-out-success.of-current.HTML").fmt(
+                branchToSlideOutName));
+        ModalityUiUtil.invokeLaterIfNeeded(ModalityState.NON_MODAL, doInUIThreadWhenReady);
+      });
     } else {
       val root = gitRepository.getRoot();
       getDeleteLocalBranchOnSlideOutGitConfigValueAndExecute(root, (@Nullable Boolean shouldDelete) -> {
         if (shouldDelete == null) {
-          ModalityUiUtil.invokeLaterIfNeeded(ModalityState.NON_MODAL,
-              () -> suggestBranchDeletion(branchToSlideOutName, doInUIThreadWhenReady));
+          ModalityUiUtil.invokeLaterIfNeeded(ModalityState.NON_MODAL, () -> suggestBranchDeletion(doInUIThreadWhenReady));
         } else {
-          handleBranchDeletionDecision(branchToSlideOutName, shouldDelete, doInUIThreadWhenReady);
```

Comment:
Also, using `branchToSlideOutName` wherever possible instead of passing it around